### PR TITLE
Added endpoints for ProductAttribute, ProductAttributeTranslation and ProductAttributeValue

### PR DESCRIPTION
**Pull request by Malin-Cebo**
| Q  | A |
| ------------- | ------------- |
| Related tickets | bagrinsergiu/brizy-cloud-sylius-demo-template/issues/127 |
| Tests |  N |

### Changelog

* New: Added endpoints for ProductAttribute, ProductAttributeTranslation and ProductAttributeValue with Subject(product) and LocaleCode filter,  Create and Update Product with ProductAttributeValue

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ProductAttributeValue.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    <resource class="%sylius.model.product_attribute_value.class%" shortName="ProductAttributeValue">
+        <attribute name="validation_groups">sylius</attribute>
+
+        <collectionOperations>
+            <collectionOperation name="admin_get">
+                <attribute name="method">GET</attribute>
+                <attribute name="path">/admin/product-attribute-values</attribute>
+                <attribute name="filters">
+                    <attribute>sylius.api.product_attribute_value_subject_filter</attribute>
+                    <attribute>sylius.api.product_attribute_value_locale_code_filter</attribute>
```
Comment (lawliet5130) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ProductAttributeValue.xml:16]: Locale filter needs to be upgraded.

**Review by lawliet5130 — CHANGES_REQUESTED** [review]

File: src/Sylius/Bundle/ApiBundle/Doctrine/Filters/TaxRateZoneFilter.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        string $operationName = null
+    ) {
+        if ($property === 'zoneCode') {
+            $queryBuilder
+                ->innerJoin('o.zone', 'zoneId')
+                ->where('zoneId.code LIKE CONCAT(\'%\', :code, \'%\')')
+                ->setParameter('code', strripos($value, '/') ? substr(strrchr($value, "/"), 1) : $value)
```
[the remark this answers is not in the document]
Comment (Malin-Cebo) [reply src/Sylius/Bundle/ApiBundle/Doctrine/Filters/TaxRateZoneFilter.php]: Fixed

File: src/Sylius/Bundle/ApiBundle/Doctrine/Filters/TaxRateZoneFilter.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        $value,
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        string $operationName = null
+    ) {
+        if ($property === 'zoneCode') {
+            $queryBuilder
+                ->innerJoin('o.zone', 'zoneId')
+                ->where('zoneId.code LIKE CONCAT(\'%\', :code, \'%\')')
```
[the remark this answers is not in the document]
Comment (Malin-Cebo) [reply src/Sylius/Bundle/ApiBundle/Doctrine/Filters/TaxRateZoneFilter.php]: Done. 
where('zoneId.code LIKE :code')

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ProductAttributeValue.xml (hunk 1 above)
Comment (Malin-Cebo) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ProductAttributeValue.xml:16]: Done. Added new filter.

File: src/Sylius/Bundle/ApiBundle/Doctrine/Filters/TaxRateZoneFilter.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        string $operationName = null
+    ) {
+        if ($property === 'zoneCode') {
+            $value = explode('/',$value);
+            $queryBuilder
+                ->innerJoin('o.zone', 'zoneId')
+                ->where('zoneId.code LIKE :code')
+                ->setParameter('code', $value[end($value)])
```
Comment (lawliet5130) [inline src/Sylius/Bundle/ApiBundle/Doctrine/Filters/TaxRateZoneFilter.php:36]:
```suggestion
                ->setParameter('code', end($value))
```

**Review by lawliet5130 — CHANGES_REQUESTED** [review]

File: src/Sylius/Bundle/ApiBundle/Doctrine/Filters/TaxRateZoneFilter.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    protected function filterProperty(
+        string $property,
+        $value,
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        string $operationName = null
+    ) {
+        if ($property === 'zoneCode') {
+            $value = explode('/',$value);
```
Comment (lawliet5130) [inline src/Sylius/Bundle/ApiBundle/Doctrine/Filters/TaxRateZoneFilter.php:32]:
```suggestion
            $value = explode('/', $value);
```

**Review by lawliet5130 — CHANGES_REQUESTED** [review]

**Review by lawliet5130 — APPROVED** [review]

**Pull request merged**